test/FsAutoComplete.Tests.Lsp/CoreTests.fs

```diff
@@ -377,51 +349,64 @@ let closeTests state =
-      Expect.isNonEmpty diags "There should be an error"
-      do! doc |> Document.close
-
-      let! diags = doc |> Document.waitForLatestDiagnostics (TimeSpan.FromSeconds 5.0)
-      Expect.isEmpty diags "There should be a final publishDiagnostics without any diags"
-    })
-    testCaseAsync "closing existing script file inside workspace doesn't clear diagnostics" (async {
-      let! (doc, diags) = server |> Server.openDocument "Script.fsx"
-      Expect.isNonEmpty diags "There should be an error"
-      do! doc |> Document.close
 
-      let! diags = doc |> Document.waitForLatestDiagnostics (TimeSpan.FromSeconds 5.0)
-      Expect.isNonEmpty diags "There should be no publishDiagnostics without any diags after close"
-    })
-    testCaseAsync "closing existing script file outside workspace clears diagnostics" (async {
-      let file = Path.Combine(root, "Script.fsx")
-      let! (doc, diags) = server |> Server.openDocument file
-      Expect.isNonEmpty diags "There should be an error"
-      do! doc |> Document.close
-
-      let! diags = doc |> Document.waitForLatestDiagnostics (TimeSpan.FromSeconds 5.0)
-      Expect.isEmpty diags "There should be a final publishDiagnostics without any diags"
-    })
+  serverTestList "close tests" state defaultConfigDto (Some workspace) (fun server ->
+    [ testCaseAsync
+        "closing untitled script file clears diagnostics"
+        (async {
 
-    testCaseAsync "closing existing file inside project & workspace doesn't clear diagnostics" (async {
-      let! (doc, diags) = server |> Server.openDocument "InsideProjectInsideWorkspace.fs"
-      Expect.isNonEmpty diags "There should be an error"
-      do! doc |> Document.close
+          let source =
+            // The value or constructor 'untitled' is not defined.
+            "let foo = untitled"
 
-      let! diags = doc |> Document.waitForLatestDiagnostics (TimeSpan.FromSeconds 5.0)
-      Expect.isNonEmpty diags "There should be no publishDiagnostics without any diags after close"
-    })
-    testCaseAsync "closing existing file inside project but outside workspace doesn't clear diagnostics" (async {
-      let file = Path.Combine(root, "InsideProjectOutsideWorkspace.fs")
-      let! (doc, diags) = server |> Server.openDocument file
-      Expect.isNonEmpty diags "There should be an error"
-      do! doc |> Document.close
-
-      let! diags = doc |> Document.waitForLatestDiagnostics (TimeSpan.FromSeconds 5.0)
-      Expect.isNonEmpty diags "There should be no publishDiagnostics without any diags after close"
-    })
-  ])
+          let! (doc, diags) = server |> Server.createUntitledDocument source
+
+          Expect.isNonEmpty diags "There should be an error"
+          do! doc |> Document.close
+
+          let! diags = doc |> Document.waitForLatestDiagnostics (TimeSpan.FromSeconds 5.0)
+          Expect.equal diags Array.empty "There should be a final publishDiagnostics without any diags"
+        })
+      testCaseAsync
```

Comment:
Do we have any need for a testCase for Task? If these suggestions aren't useful, feel free to disregard.

Reply:
You mean like `testCaseTask` ?